src/store/reducers/contact.reducer.ts

```diff
@@ -11,15 +11,24 @@ import {
   DELETE_CONTACT_SUCCESS,
   GET_CONTACTS_FAILED,
   GET_CONTACTS_SUCCESS,
+  UPDATE_CONTACT_FAILED,
+  UPDATE_CONTACT_SUCCESS,
 } from '../../types/store/contact.action.types';
 import {IContactState} from '../../types/store/contact.types';
+import {sortBy, SortOrder} from '../../utils/SortUtil';
 
 const initialState: IContactState = {
   loading: false,
   contacts: [],
 };
 
 const contactReducer = (state: IContactState = initialState, action: ContactActionTypes): IContactState => {
+  if ('payload' in action) {
```

Comment:
this will not work as some payloads are singular, so you first need to add the payload (if it's a contact) to the contacts and then assign it to the state

Reply:
ALso what happends if the payload is not a contact or contacts?

Reply:
I think code has been miss-read here. Please have a look again. 
I am altering the param. And I have seen it work at least for add and update contact.
I think this is centeral enough place and there is a check whether the payload is Array or not.
I will test other cases as well.

Reply:
delete also works.
Get also works

Reply:
the problem is that you sort if it is an array, but then add the singular contact to the sorted array, which i think is then not sorted properly anymore